### DISCUSSION TITLE
Archive related question and activating Year, Month and Day...

### DISCUSSION THREAD

**Discussion by benibilme** [mmistakes/minimal-mistakes#2469] [Q&A]
Hello, 

I am newbie in web development and I am trying to learn jekyll and to build a personal site for myself.I have had some experience with ruby and am comfortable with ruby ecosystem. I have read all jekyll and minimal-mistakes theme documentation. On the other hand, I have difficulty in implementing several things in my website, specifically I can not get archives working for year, month and day. I have been stuck at year, to which all others depend upon.  I have also have problem with the home and blogging url but I will ask this in another issue... 

My main menu currently is as below. I want root / to be my splash screen (if possible, I would like to show latest several blogs on it), but in general I want all my blogs to be accesible via url /blog.  I want all the archives to be accessible through /archives/  like /archives/categories, /archives/tags/, /archives/year, /archives/year/month, /archive/year/month/day, to save horizontal space at the top.  From the issues of minimal-mistakes, I have learned that embedded menus like archives>categories etc. are not possible. So I have made a separate page for /archives/ which provides links to categories, tags and years (for now) on it. 

```
main:
  - title: "About Me"
    url: /about/
  - title: "Blog"
    url: /blog/
  - title: "Archives"
    url: /archives/
```

I have blindly applied following in _config without understanding internals. 
```
# jekyll archieve related settings
category_archive:
  type: jekyll-archives
  path: /archives/categories/
tag_archive:
  type: jekyll-archives
  path: /archives/tags/
year_archive:
  type: jekyll-archives
  path: /archives/years/ 
jekyll-archives:
  enabled:
    - categories
    - tags
    - years
  layouts:
    category: 'archive-taxonomy'
    tag:      'archive-taxonomy'
    year:     'archive-taxonomy'
  permalinks:
    category: '/archives/categories/:name/'
    tag:      '/archives/tags/:name/'
    year:     '/archives/years/:name/'

```
To the best of my understanding, I need to create there pages for /archives/categories, /archives/tags and /archives/years, and I created them based on the sample provided in minimal-mistakes web site. 
The settings for archiving of categories and tags work regardless of the existence of jekyll-archive plugin (which is mentioned at the documentation because of liquid). However year is not working. 
I also created pages for single category, tag and year.  The files for years, single year are presented below, which they show nothing when accessed. 
I am planning to use asciidoc so I have not been able to convert all files to asciidoc yet for this question only /archives/ page is in asciidoc and it is working. 

```
***********************
archive.adoc
= Site Archive
:page-layout: single
:page-permalink: /archives/

* link:/archives/categories[Posts by Categories] 
* link:/archives/tags[Posts by Tags] 
* link:/archives/years[Posts by Years]
**********************
archive-years.md
---
title: "Years"
layout: archive
permalink: /archives/years/
---
*********************
archive-single-year.md*
---
title: "Years"
layout: archive
permalink: /archives/years/
---
***********************
(These files conforms what you have provided in the documentation, and they are working) 
archive-categories.md
archive-single-category.md
archive-single-tag.md
archive-tags.md
```
I  appreciate any help..

**Comment by benibilme** [comment]
Hello,

Year archive has started to work with following _config setting and chaging the layout to posts in archive-years.md file. Could you please identify how to handle month and days if possible...

```
category_archive:
  type: jekyll-archives
  path: /archives/categories/
tag_archive:
  type: jekyll-archives
  path: /archives/tags/
year_archive:
  type: jekyll-archives
  path: /archives/years/ 
jekyll-archives:
  enabled:
    - categories
    - tags
    - years
  layouts:
    category: 'category'
    tag:      'tag'
    year:     'posts'
  permalinks:
    category: '/archives/categories/:name/'
    tag:      '/archives/tags/:name/'
    year:     '/archives/years/:name/'
```

```
---
archive-years.md
title: "Years"
layout: posts
permalink: /archives/years/
---
```

**Comment by mmistakes** [comment]
Not sure it’s possible to do months and days without customizing the included layouts. I’ve only built layouts to do taxonomies (category/tag), collections, and posts by year.

You’ll want to consult jekyll-archives documentation and [sample code](https://github.com/jekyll/jekyll-archives/blob/master/docs/layouts.md) to see what modifications you’d need to make for months/days.

**Comment by benibilme** [comment]
Hello,

Thank you for the response...Even though I have read the full theme  and jekyll-archive documentation truly , maybe because my inadequate level of knowledge, I have not been able to get categories, tags and years archieve working with jekyll-archive. 

I create a new site, quit using jekyll-archive, just depending on liquid with updated, following settings, now categories, tags and year archives are working. 

```
more _data/navigation.yml 
main:
  - title: "About Me"
    url: /about/
  - title: "Blog"
    url: /blog/
  - title: "Categories"
    url: /categories/
  - title: "Tags"
    url: /tags/
  - title: "Years"
    url: /years/
```
Relevant settings in _config.yml

```
category_archive:
  type: liquid
  path: /categories/
tag_archive:
  type: liquid
  path: /tags/
year_archive:
  type: liquid
  path: /years/
```
 Relevant files 
```
more _pages/posts-by-category.adoc 
= Posts by Category
:page-layout: categories
:page-permalink: /categories/

more _pages/posts-by-tag.adoc 
= Posts by Tags
:page-layout: tags
:page-permalink: /tags/

more _pages/posts-by-year.adoc 
= Years
:page-layout: posts
:page-permalink: /years/
```

**However I believe that the documentation must be improved with following points for the novice.** 

1. Which archive functions  are supported of out the box?  (You stated here that, month, days are not supported, which is not stated clearly in the documentation) 
2. The correct settings for the default/liquid  as well as jekyll-archive for categories, tags and years...(categories, tags are provided, years are not) Someone has to dig that. For example, there are corresponding layouts for categories and tags but not for years. 
3.  Explicit statements and sample files that must be created by the user when using jekyll-archive, liquid etc. 
4. The sample files for edge-cases are misleading for novice people for me...They should be either removed and enhanced with sample layouts. 

Thank you again for this nice theme and great work.

**Comment by benibilme** [comment]
Hello, because of mostly multi language issues, I have decided to change platform. I have switched to Hugo. Thank you for the beautiful theme and the support...
